DB/src/main/java/io/deephaven/db/v2/locations/parquet/topage/ToPageWithDictionary.java

```diff
@@ -44,59 +46,61 @@ public final ChunkType getChunkType() {
 
     @Override
     @NotNull
-    public final Object getResult(final ColumnPageReader columnPageReader) throws IOException {
+    public final Object getResult(@NotNull final ColumnPageReader columnPageReader) throws IOException {
         if (columnPageReader.getDictionary() == null) {
             return ToPage.super.getResult(columnPageReader);
         }
-        int[] keys = new int [columnPageReader.numValues()];
-        IntBuffer offsets = columnPageReader.readKeyValues(IntBuffer.wrap(keys), NULL_INT);
+
+        final int[] keys = new int[columnPageReader.numValues()];
+        final IntBuffer offsets = columnPageReader.readKeyValues(IntBuffer.wrap(keys), NULL_INT);
 
         return offsets == null ? keys : new DataWithOffsets(offsets, keys);
     }
 
     @Override
     @NotNull
-    public final T[] convertResult(final Object result) {
+    public final DATA_TYPE[] convertResult(@NotNull final Object result) {
         if (!(result instanceof int[])) {
             return convertResultFallbackFun.apply(result);
         }
-        int[] from = (int []) result;
+
+        final int[] from = (int[]) result;
         //noinspection unchecked
-        T[] to = (T[])Array.newInstance(nativeType, from.length);
+        final DATA_TYPE[] to = (DATA_TYPE[]) Array.newInstance(nativeType, from.length);
 
-        for (int i = 0; i < from.length; ++i) {
-            to[i] = dictionary.get(from[i]);
+        for (int ii = 0; ii < from.length; ++ii) {
+            to[ii] = dictionary.get(from[ii]);
         }
 
         return to;
     }
 
     @Override
     @NotNull
-    public final ObjectChunk<T, ATTR> getDictionary() {
+    public final ObjectChunk<DATA_TYPE, ATTR> getDictionary() {
         return dictionary.getChunk();
     }
 
     @NotNull
-    public final StringSetImpl.ReversibleLookup<T> getReversibleLookup() {
+    public final StringSetImpl.ReversibleLookup<DATA_TYPE> getReversibleLookup() {
         return dictionary;
     }
 
     @Override
     @NotNull
-    public final ToPage<Attributes.DictionaryKeys, int[]> getDictionaryKeysToPage() {
-        return new ToPage<Attributes.DictionaryKeys, int[]>() {
```

Comment:
Long story, but totally worth explaining. It was a bit of a trade-off.

The short answer is: because that's what `convertResult` returns now.

Most of the code that uses `getDictionaryKeysToPage` is just looking to get the `int[]` returned by `getResult` (as an `Object`) and convert it to the intended upward-facing data type before wrapping that in a chunk and caching it. The exception is the symbol table code up in `RegionedColumnSourceWithDictionary`. We expose `long` ids that live in the same region address space they came from, but with their sub-region offsets within the dictionary's space.

For example, in a nested structure, maybe I'm in outer region 20, inner region 5 (out of 10, so 4 bits of address space), and the dictionary offset is 1911, so the resulting symbol id is:
```
20 << 43 | 5 << 39 | 1911
178670639515511
```
So, to implement this, I need to get `1911` and its friends from the lower level, and then OR on the higher levels' region offsets. The easiest way is to fill a chunk of longs from the bottom (dictionary) region, and than manipulate it in-place.

Note that (`SoftReference`-based) caching built into our Parquet regions is on the materialized results (so after `convertResult`, not `getResult`). I decided it made more sense to convert to longs and cache that, so that my `fillChunk` looks like an array copy, than to cache the ints and have to do a row-by-row cast when retrieving values. This also avoided adding yet-another wrapper page.

Reply:
Rewriting the int[] to long[] is sad, but I guess such is life.